### ISSUE TITLE
Version 1.10.29 breaks EC2 searches with tags (by specifying MaxResults)

### ISSUE THREAD

**Issue by hashbrowncipher** [aws/aws-cli#1976]
The following works in version 1.10.28:

    aws ec2 describe-instances --region=us-west-1 --filters "Name=tag:aws:autoscaling:groupName,Values=consul"

It does not work in 1.10.29. Instead, this result is returned:

    A client error (InvalidParameterCombination) occurred when calling the DescribeInstances operation: The parameter 'maxResults' cannot be used with tag filters in the parameter 'filterSet'. Remove either the tag filters from 'filterSet' or the 'maxResults' parameter and try again.

**Comment by leo100** [comment]
Broken with: aws-cli/1.10.29 Python/2.6.9 Linux/4.4.8-20.46.amzn1.x86_64 botocore/1.4.20

# aws ec2 describe-volumes --volume-ids vol-bce02671

A client error (InvalidParameterCombination) occurred when calling the DescribeVolumes operation: The parameter volumeSet cannot be used with the parameter maxResults

# Or trying to mount Ebs volume produces:
A client error (InvalidParameterCombination) occurred when calling the DescribeVolumes operation: The parameter volumeSet cannot be used with the parameter maxResults
Volume ID: vol-bce02671 State:
Volume vol-bce02671 is in  and not available , exiting

But works with: 
aws-cli/1.10.28 Python/2.6.9 Linux/4.4.8-20.46.amzn1.x86_64 botocore/1.4.19

**Comment by hieurl** [comment]
Same problem here
Suddenly many of my cron jobs failed today after upgrading awscli

**Comment by watsonrp** [comment]
Me too

`aws ec2 describe-instances --instance-ids i-AAAAAAAA`

Produces...
```

A client error (InvalidParameterCombination) occurred when calling the DescribeInstances operation:
The parameter instancesSet cannot be used with the parameter maxResults
```

**Comment by sue445** [comment]
me too 😢 

```sh
$ aws --version
aws-cli/1.10.29 Python/2.7.5 Linux/3.10.0-327.10.1.el7.x86_64 botocore/1.4.20

$ aws ec2 describe-instances --instance-ids "xxxxxxxxxx" --region "ap-northeast-1"

A client error (InvalidParameterCombination) occurred when calling the DescribeInstances operation: The parameter instancesSet cannot be used with the parameter maxResults

$ echo $?
255
```

**Comment by onuki-yoshihisa** [comment]
me too.

```
# aws ec2 describe-instances --region ${REGION} --instance-ids ${INSTANCE_ID} 

A client error (UnauthorizedOperation) occurred when calling the DescribeInstances operation: You are not authorized to perform this operation.
```

**Comment by jamesls** [comment]
Change that's causing this issue has been reverted: https://github.com/aws/aws-cli/pull/1977

**Comment by tsyardley** [comment]
can confirm this has broken using filters in describe-instances

**Comment by jamesls** [comment]
We've just released v1.10.30 of the CLI which reverts the change that caused this issue.  Please upgrade to 1.10.30.

Going to go ahead and close out this issue.  Let us know if anyone's still seeing issues on v1.10.30.

**Comment by nswanson7** [comment]
I'm seeing this error using the get_only_instances() function with boto.
